iSeg/LoaderWidgets.cpp

```diff
@@ -1254,80 +979,169 @@ LoaderColorImages::LoaderColorImages(SlicesHandler* hand3D, eImageType typ, std:
 	xoffset->setValue(0);
-	hbox3->show();
-	hbox4 = new Q3HBox(vbox2);
-	xl = new QLabel(QString("x-Length: "), hbox4);
-	xlength = new QSpinBox(0, 2000, 1, hbox4);
-	xlength->show();
-	xlength->setValue(256);
-	yl = new QLabel(QString("y-Length: "), hbox4);
-	ylength = new QSpinBox(0, 2000, 1, hbox4);
-	ylength->show();
-	ylength->setValue(256);
-	hbox4->show();
-	vbox2->show();
-	hbox2->show();
-	hbox6 = new Q3HBox(vbox1);
-	loadFile = new QPushButton("Open", hbox6);
-	cancelBut = new QPushButton("Cancel", hbox6);
-	hbox6->show();
 
-	vbox1->show();
-	setSizePolicy(QSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed));
+	auto xl = new QLabel(QString("x-Length: "));
+	xlength = new QSpinBox(0, 2000, 1, nullptr);
+	xlength->setValue(256);
 
-	vbox1->setFixedSize(vbox1->sizeHint());
+	auto yl = new QLabel(QString("y-Length: "));
+	ylength = new QSpinBox(0, 2000, 1, nullptr);
+	ylength->setValue(256);
 
-	subsect_toggled(subsect->isChecked());
+	auto subsect_layout = new QGridLayout(2,4);
+	subsect_layout->addWidget(xoffs);
+	subsect_layout->addWidget(xoffset);
+	subsect_layout->addWidget(xl);
+	subsect_layout->addWidget(xlength);
+	subsect_layout->addWidget(yoffs);
+	subsect_layout->addWidget(yoffset);
+	subsect_layout->addWidget(yl);
+	subsect_layout->addWidget(ylength);
+	auto subsect_options = new QWidget;
+	subsect_options->setLayout(subsect_layout);
+
+	loadFile = new QPushButton("Open");
+	cancelBut = new QPushButton("Cancel");
+	auto button_layout = new QHBoxLayout;
+	button_layout->addWidget(loadFile);
+	button_layout->addWidget(cancelBut);
+	auto button_row = new QWidget;
+	button_row->setLayout(button_layout);
+
+	auto top_layout = new QVBoxLayout;
+	top_layout->addWidget(map_to_lut);
+	top_layout->addWidget(subsect);
+	top_layout->addWidget(subsect_options);
+	top_layout->addWidget(button_row);
+	setLayout(top_layout);
+	setMinimumSize(150,200);
+
+	map_to_lut_toggled();
 
 	QObject::connect(loadFile, SIGNAL(clicked()), this, SLOT(load_pushed()));
 	QObject::connect(cancelBut, SIGNAL(clicked()), this, SLOT(close()));
-	QObject::connect(subsect, SIGNAL(clicked()), this, SLOT(subsect_toggled()));
+	QObject::connect(map_to_lut, SIGNAL(clicked()), this, SLOT(map_to_lut_toggled()));
 }
 
-LoaderColorImages::~LoaderColorImages() { delete vbox1; }
+LoaderColorImages::~LoaderColorImages() {}
 
-void LoaderColorImages::subsect_toggled(bool isset)
+void LoaderColorImages::map_to_lut_toggled()
 {
-	if (isset)
+	// enable/disable
+	subsect->setEnabled(!map_to_lut->isChecked());
+}
+
+void LoaderColorImages::load_pushed()
+{
+	if (map_to_lut->isChecked())
 	{
-		vbox2->show();
+		load_quantize();
 	}
 	else
 	{
-		vbox2->hide();
+		load_mixer();
 	}
 }
 
-void LoaderColorImages::subsect_toggled()
+void LoaderColorImages::load_quantize()
 {
-	bool isset = subsect->isOn();
-	if (isset)
+	QString filename = QFileDialog::getOpenFileName(QString::null,
+		"iSEG Color Lookup Table (*.lut *.h5)\nAll(*.*)", this);
```

Comment:
If you want translation later on. There is a tr() function (http://doc.qt.io/qt-5/i18n-source-translation.html)

Reply:
ok, thanks